### PR TITLE
Documentation: Creating Examples

### PR DESCRIPTION
**Pull request by gregwht**
Created examples for:
MouseDown()
MouseX()
MouseY()
ASREnvelope()
Line()
Impulse()
ClockDivider()
Sequence()

And fixing a typo in BufferPlayer().

### CODE REVIEW

File: docs/library/sequencing/clockdivider/example-0.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+oscillator_a = TriangleOscillator(220)
+oscillator_b = TriangleOscillator(440)
+
+envelope_a = ASREnvelope(0.01, 0.0, 0.25, 1.0, clock)
+envelope_b = ASREnvelope(0.01, 0.0, 0.5, 1.0, divided_clock)
+
+voice_a = oscillator_a * envelope_a
+voice_b = oscillator_b * envelope_b
+
+left = StereoPanner(voice_a, -1.0)
```
Comment (ideoforms) [inline docs/library/sequencing/clockdivider/example-0.py:19]: I love this example!
To be slightly pedantic again - this formulation of "voice_a through left channel and voice_b through right channel" could (slightly more efficiently) be implemented using `ChannelArray(voice_a, voice_b)`. But I like the use of `StereoPanner` here, so maybe you could do `StereoPanner(voice_a, -0.5)` and `StereoPanner(voice_b, 0.5)` to do something that couldn't be achieved with a ChannelArray.

Comment (ideoforms) [reply docs/library/sequencing/clockdivider/example-0.py:19]: Also missing a graph.wait()?

File: docs/library/sequencing/sequence/example-0.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from signalflow import *
+graph = AudioGraph()
+
+#-------------------------------------------------------------------------------
+# Creating a sequence using the MIDI note values of a C Major scale, starting on middle C.
+#-------------------------------------------------------------------------------
+clock = Impulse(2.0)
+sequence = Sequence([ 60, 62, 64, 65, 67, 69, 71, 72 ], clock)
```
Comment (ideoforms) [inline docs/library/sequencing/sequence/example-0.py:8]: Another nice example!

File: docs/library/stochastic/randomchoice/index.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ description: RandomChoice: Pick a random value from the given array. If a clock
 # RandomChoice
 
 ```python
-RandomChoice(values=std::vector<float> ( ), clock=None, reset=None)
+RandomChoice(values=std : : vector <float >(), clock=None, reset=None)
```
Comment (ideoforms) [inline docs/library/stochastic/randomchoice/index.md:9]: Some auto-formatting is causing these colons to be broken up, which shouldn't happen. I remember seeing this before but can't remember the cause. Can you merge your branch with latest `master` and see if it's still happening?

**Review by ideoforms — CHANGES_REQUESTED** [review]
Really nice examples, and thanks for the improvements to the docs! I've added a few comments inline.

File: docs/library/envelope/line/example-0.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from signalflow import *
+graph = AudioGraph()
+
+#-------------------------------------------------------------------------------
+# Using a line to control the gain of an oscillator, emulating a sidechain ducking effect.
+#-------------------------------------------------------------------------------
+clock = Impulse(frequency=1.0)
+line = Line(0.0, 0.5, 1.0, 0, clock)
```
[the remark this answers is not in the document]
Comment (gregwht) [reply docs/library/envelope/line/example-0.py:8]: Good idea re: `False` -- should I update that in the documentation to match? Currently the example says `loop=0`

File: docs/library/sequencing/clockdivider/example-0.py (hunk 1 above)
Comment (gregwht) [reply docs/library/sequencing/clockdivider/example-0.py:19]: Great catch re: graph.wait()!
Good idea re: StereoPanner -- I'll change the pan values to -0.75 and 0.75 to give a bit more stereo separation, which should allow for easy distinction between the two voices whilst still doing something ChannelArray can't. Feel like with -0.5/0.5 the distinction is less clear and doesn't quite demonstrate the ClockDivider as effectively, if you get me?

File: docs/library/stochastic/randomchoice/index.md (hunk 3 above)
Comment (gregwht) [reply docs/library/stochastic/randomchoice/index.md:9]: Having trouble with merging my branch to latest `master` -- might be worth going through this together on Friday.

**Review by ideoforms — APPROVED** [review]
💯

**Pull request merged**